File: chakhaeng-backend/src/main/java/com/ssafy/chakeng/gamification/domain/Mission.java
package com.ssafy.chakeng.gamification.domain;

import com.ssafy.chakeng.user.domain.User;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Mission {
    @Id
    private String id;

    private String title;
    private String description;
    private String iconUrl;
    private boolean isCompleted;

    private int currentProgress;
    private int targetProgress;
    private String rewardName;

    @ManyToOne(fetch = FetchType.LAZY)
    private User user;
}

File: chakhaeng-android/app/src/main/java/com/sos/chakhaeng/domain/model/profile/Mission.kt
package com.sos.chakhaeng.domain.model.profile

import com.sos.chakhaeng.R

data class Mission(
    val id: String,
    val title: String,
    val description: String,
    val iconRes: Int? = R.drawable.badge_safety,
    val isCompleted: Boolean,
    val currentProgress: Int = 0,
    val targetProgress: Int = 1,
    val rewardName: String = ""
) {
    val progressPercentage: Float
        get() = if (targetProgress > 0) {
            (currentProgress.toFloat() / targetProgress.toFloat() * 100f).coerceAtMost(100f)
        } else 100f
}